2024.2 (#31289)
home-assistant/home-assistant.github.io: source/_posts/2024-02-07-release-20242.markdown

---
layout: post
title: "2024.2: More voice, more icons, more integrations, more... everything!"
description: "Improved drag 'n drop in automations, proximity, custom Assist responses, export your data, ZHA firmware updates, Matter diagnostics and actions, icons everywhere, and more!"
date: 2024-02-07 00:00:00
date_formatted: "Febuary 7, 2024"
author: Franck Nijhof
author_twitter: frenck
comments: true
categories:
  - Release-Notes
  - Core
og_image: /images/blog/2024-02/social.png
---

<a href='/integrations/#version/2024.2'><img src='/images/blog/2024-02/social.png' style='border: 0;box-shadow: none;'></a>

Home Assistant Core 2024.2! 🥰

Undoubtedly, you've heard about last year's ["The Year of the Voice"](/blog/2022/12/20/year-of-voice/).
2023 might be over, but we are definitely not done with voice yet! This release
contains some very cool new features for voice.

> Not just that, we've also launched a [voice assistant contest] that you could

Reviewer: Link missing for contest

## All known assist devices

Last addition for voice, we've added a small additional page to give you an
overview of all the Assist devices you have active in your Home. You can find
this on the {% my voice_assistants title="Assist configuration dashboard" %}
by selecting the new "assist devices" button.

<img class="no-shadow" src='/images/blog/2024-02/assist-devices.png' alt='Screenshot showing a custom response in Assist.'>

Or, use this My Home Assistant button below to navigate to the voice assistants
configuration dashboard:

{% my voice_assistants badge %}

## Updating your Zigbee devices

[ZHA](/integrations/zha) (Zigbee Home Automation, our built-in Zigbee integration)
now provides update entities to Home Assistant!

This means you can get notified when a device has a firmware update available,
and you can trigger the update from Home Assistant!

<img class="no-shadow" src='/images/blog/2024-02/zha-updating-devices.png' alt='Screenshot showing an Zigbee device connected via ZHA being updated.'>

Noticed the weird version numbers? Unfortunately, that is how version numbers

> in the Zigbee standard work and look. We haven't figured out away to translate

Reviewer 2: ```suggestion
in the Zigbee standard work and look. We haven't figured out a way to translate
```

all issues with your system.

<img class="no-shadow" src='/images/blog/2024-02/repair-reauth.png' alt='Screenshots showing a repair issue being raised for an integration that needs re-authentication.'>

## Revamped Proximity integration

For this release, [@mib1185] completely revamped our [Proximity] integration.
In case you are unfamiliar with it, it allows you to monitor the proximity of
persons to a particular [zone]. It provides information on how close one is
to a zone and if they are traveling towards or away from it.

It is an extremely powerful tool for automations. It allows you to create
automations based on the proximity of people. For example, if you are within
a certain range from your home and traveling towards it, you could use
Home Assistant to turn on the lights, open the garage door, turn up the heating,
and start your favorite playlist in the living room. Welcome home! 🤗

The best part? This integration is now available to set up from the UI!
Not just that, but it now uses normal sensor entities that we are all already
familiar with. This will make automating and displaying the data a breeze!

> <img class="no-shadow" src='/images/blog/2024-02/proximity.png' alt='Screenshots of an set up proximity integration, the proximity of a person towards the home zone.'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2024-02/proximity.png' alt='Screenshots of a set up proximity integration, and the proximity of a person moving towards the home zone.'>
```

  entity, you will now have the option to invert its behavior.
  Thanks, [@emontnemery]!
- The [Ecovacs] integration received lots of love from [@edenhaus] and now
  has support for many Deebot vacuum models with tons of entities to automate
  with. Nice job!
- [@mib1185] added automatic cleanups of old/unused refresh tokens 🧹. This
  means you will be logged out on a device that hasn't used Home Assistant for
  90 days. Clean and tidy!
- The [Tuya] integration can now be set up directly using the Tuya Smart
  or Smart Life app. No more need for developer accounts or complicated
  procedures!
- Using the [utility meter] integration on, for example, your solar inverter,
  which turns off at night? You can now set the sensor to always be available,
  even when the source entity is unavailable. Thanks, [@dougiteixeira]!
- The [TP-Link Smart Home] integration now supports newer devices,
  including [Tapo-branded] devices! Awesome job [@sdb9696], [@bdraco],
  and [@rytilahti]!
- [@AngellusMortis] added new sound event sensors introduced in
  UniFi Protect version 2.11 to the [UniFi Protect] integration. Nice!

> - Are you an Jinja templating Ninja? [@TNTLarsn] added a new `bitwise_xor`

Reviewer 2: ```suggestion
- Are you a Jinja templating Ninja? [@TNTLarsn] added a new `bitwise_xor`
```

## Need help? Join the community!

Home Assistant has a great community of users who are all more than willing
to help each other out. So, join us!

Our very active [Discord chat server](/join-chat) is an excellent place to be
at, and don't forget to join our amazing [forums](https://community.home-assistant.io/).

Found a bug or issue? Please report it in our [issue tracker](https://github.com/home-assistant/core/issues),
to get it fixed! Or, check [our help page](/help) for guidance for more
places you can go.

Are you more into email? [Sign-up for our Building the Open Home Newsletter](/newsletter)
to get the latest news about features, things happening in our community and
other news about building an Open Home; straight into your inbox.

## Backward-incompatible changes

{% details "Authentication" %}

We changed the way refresh tokens are handled. Refresh tokens are part of what
keeps you logged in to Home Assistant.

Refresh tokens will be automatically deleted when unused. A refresh token
is considered unused if it has not been used for a login within 90 days.

If your logged-in session hasn’t been used for 90 days, you will be logged out.
You will need to log in again.

> We recommend using [Long-lived access tokens](/docs/auth_api/#long-lived-access-token)

Reviewer 3: Same here, points to normal/missing docs, not the developer docs.

([@frenck] - [#108163]) ([documentation](/integrations/cisco_webex_teams))

[@frenck]: https://github.com/frenck
[#108163]: https://github.com/home-assistant/core/pull/108163

{% enddetails %}

{% details "EZVIZ" %}

The services `ezviz.ptz`, `ezviz.sound_alarm`, and `alarm_sound` have been moved
to button and select entities in release 2023.8. Support for the deprecated
services has now been completely removed.

If you are using these services in automations or scripts, you will need to
update them to match this change.

([@jpbede] - [#107582]) ([documentation](/integrations/ezviz))

[@jpbede]: https://github.com/jpbede
[#107582]: https://github.com/home-assistant/core/pull/107582

{% enddetails %}

{% details "Google Generative AI Conversation" %}

Any previous integration setups might have to be recreated or reconfigured with
the new `models/gemini-pro` instead of the legacy `models/chat-bison-001`.

([@tronikos] - [#105789]) ([documentation](/integrations/google_generative_ai_conversation))

[@tronikos]: https://github.com/tronikos
[#105789]: https://github.com/home-assistant/core/pull/105789

{% enddetails %}

{% details "HomeWizard Energy" %}

> The Gas meter ID sensor is removed andthe meter ID is now shown as a serial

Reviewer 2: ```suggestion
The Gas meter ID sensor is removed and the meter ID is now shown as a serial
```

[@jbouwh]: https://github.com/jbouwh
[#107199]: https://github.com/home-assistant/core/pull/107199

---

When an MQTT entity has a device name equal to the entity name or starts with
the device name, this will no longer be corrected.

When entities like these are created, they will have device and entity names in
the friendly name. Suppose the device name is omitted because it is the same
as the entity name. In that case, the entity name attribute should be null
in the JSON payload or YAML configuration.

[Related developer blogpost](https://developers.home-assistant.io/blog/2023/07/21/change-naming-mqtt-entities)

([@jbouwh] - [#107188]) ([documentation](/integrations/mqtt))

[@jbouwh]: https://github.com/jbouwh
[#107188]: https://github.com/home-assistant/core/pull/107188

---

The support for MQTT vacuum entities with a `legacy` schema is removed after 6
months of deprecation. Users can use the `state` schema instead,
which is now the default.

([@jbouwh] - [#107274]) ([documentation](/integrations/mqtt))

[@jbouwh]: https://github.com/jbouwh
[#107274]: https://github.com/home-assistant/core/pull/107274

{% enddetails %}

{% details "Proximity" %}

> The proximity entity (`proximit.*`) is deprecated and will be removed in 2024.8.

Reviewer 2: ```suggestion
The proximity entity (`proximity.*`) is deprecated and will be removed in 2024.8.
```

([@mib1185] - [#108730]) ([documentation](/integrations/proximity))

[@mib1185]: https://github.com/mib1185
[#108730]: https://github.com/home-assistant/core/pull/108730

{% enddetails %}

{% details "Renson" %}

The start time of day and start time of night sensors have been replaced by
time entities. If you used these sensors in automations or scripts, you will
need to update them to use the new time entities.

([@jimmyd-be] - [#105031]) ([documentation](/integrations/renson))

[@jimmyd-be]: https://github.com/jimmyd-be
[#105031]: https://github.com/home-assistant/core/pull/105031

{% enddetails %}

{% details "RESTful Command" %}

Call to RESTful command services will no longer silently fail and will raise an
exception on, for example, timeout or decoding errors.

You could consider using [`continue_on_error`](/docs/scripts/#continuing-on-error)
for scripts and automations that use RESTful commands that are allowed to fail
occasionally.

([@RoboMagus] - [#97208]) ([documentation](/integrations/rest_command))

[@RoboMagus]: https://github.com/RoboMagus
[#97208]: https://github.com/home-assistant/core/pull/97208

{% enddetails %}

{% details "Roborock" %}

> The services `vacuum.start_pause` was deprecated in 2023.8. Support for the

Reviewer 2: ```suggestion
The service `vacuum.start_pause` was deprecated in 2023.8. Support for the
```

> deprecated services has now been completely removed.

Reviewer 2: ```suggestion
deprecated service has now been completely removed.
```

> If you are still using them, please adjust your automations and scripts,

Reviewer 2: ```suggestion
If you are still using this service, please adjust your automations and scripts,
```

> use `vacuum.pause` or `vacuum.start` instead.

Reviewer 2: ```suggestion
to use `vacuum.pause` or `vacuum.start` instead.
```

calls the `homeassistant.update_entity` service.

([@bdraco] - [#104208]) ([documentation](/integrations/tplink))

[@bdraco]: https://github.com/bdraco
[#104208]: https://github.com/home-assistant/core/pull/104208

{% enddetails %}

{% details "Traccar" %}

The YAML configuration for the Traccar integration has been deprecated
and will be removed in Home Assistant 2024.8.0. Your YAML configuration will
be automatically imported to the new [Traccar Server](/integrations/traccar_server)
integration.

If you previously used the `scan_interval` option, this is not being imported.
If you are in need of a custom polling interval, you can use an automation
to define your own custom polling behavior.
[Learn more about that here.](/common-tasks/general/#defining-a-custom-polling-interval)

([@ludeeus] - [#109226]) ([documentation](/integrations/traccar_server))

[@ludeeus]: https://github.com/ludeeus
[#109226]: https://github.com/home-assistant/core/pull/109226

{% enddetails %}

{% details "Tuya" %}

Tuya has provided an easier and improved login method for Home Assistant users.

Having a developer account with Tuya is no longer required; instead, you can

> scan a QR code with your Tuya Smart of Smart Life app to authenticate it

Reviewer 2: ```suggestion
scan a QR code with your Tuya Smart or Smart Life app to authenticate it
```
